pkg/apiserver/certificate/certificate.go

```diff
@@ -39,8 +39,13 @@ var (
 	// certReadyTimeout is the timeout we will wait for the TLS Secret being ready. Declaring it as a variable for testing.
 	certReadyTimeout = 2 * time.Minute
 
-	// globalMaxRotateDuration forces all rotations to occur within a year minus a day.
-	globalMaxRotateDuration = time.Hour * (24 * 364)
+	// defaultMaxRotateDuration forces all rotations to occur within half the default expiration time.
+	// This is set to half the duration of the default expiration returned from the MaybeDefaultWithSelfSignedCerts
+	// secure serving options from k8s.io/apiserver, which uses 1 year self-signed certificates by default.
+	//
+	// See https://github.com/kubernetes/client-go/blob/e05ca24c4895e2d52218a1b1dbaa84f21459b632/util/cert/cert.go#L95-L96
+	// for how the default maxAge is set.
+	defaultMaxRotateDuration = time.Hour * (24 * 183)
```

Comment:
Maybe this is too aggressive? Thoughts on 1 month before expiration instead?

Reply:
I think half of the expiration date is not bad. But maybe this should be done in `nextRotationDuration`?
By replacing `duration := x509Cert.NotAfter.Sub(time.Now()) - time.Hour*24` with `duration := x509Cert.NotAfter.Sub(time.Now()) / 2`

Right now I believe this constant was mostly meant to be used for testing the rotation implementation.

Reply:
> Right now I believe this constant was mostly meant to be used for testing the rotation implementation.

Ah, this explains why it existed in two places. Will update shortly, thanks!

Reply:
Updated, thanks!